webportal/src/app/market_list/components/item_list.jsx

```diff
@@ -50,11 +59,110 @@ const ItemList = props => {
             {isNil(type) ? 'All' : capitalize(type)}
           </Text>
           <HorizontalLine />
-          <GridWrapper>
-            {itemList.map(item => (
-              <ItemCard key={item.id} item={item} />
-            ))}
-          </GridWrapper>
+          <Section>
+            <Stack gap='s1'>
+              <Text variant={'large'}>Official</Text>
+              {!isEmpty(
+                items.filter(item => {
+                  return item.author === 'OpenPAI';
+                }),
+              ) && (
+                <div>
+                  <GridWrapper>
+                    {isPartialOfficial
+                      ? items
+                          .filter(item => {
+                            return item.author === 'OpenPAI';
+                          })
+                          .slice(0, 6)
+                          .map(item => <ItemCard key={item.id} item={item} />)
+                      : items
+                          .filter(item => {
+                            return item.author === 'OpenPAI';
+                          })
+                          .map(item => <ItemCard key={item.id} item={item} />)}
+                  </GridWrapper>
+                  {isPartialOfficial && (
+                    <ActionButton
+                      onClick={() => {
+                        setIsPartialOfficial(false);
+                      }}
+                    >
+                      More
+                    </ActionButton>
+                  )}
+                </div>
+              )}
+            </Stack>
+          </Section>
+          <Section>
+            <Text variant={'large'}>Other Public</Text>
+            {!isEmpty(
+              items.filter(item => {
+                return item.author !== 'OpenPAI' && item.isPublic;
+              }),
+            ) && (
+              <div>
+                <GridWrapper>
+                  {isPartialPublic
+                    ? items
+                        .filter(item => {
+                          return item.author !== 'OpenPAI' && item.isPublic;
+                        })
+                        .slice(0, 6)
+                        .map(item => <ItemCard key={item.id} item={item} />)
+                    : items
+                        .filter(item => {
+                          return item.author !== 'OpenPAI' && item.isPublic;
+                        })
+                        .map(item => <ItemCard key={item.id} item={item} />)}
+                </GridWrapper>
```

Comment:
I don't know well about JS or REACT, but looks to me that you can create a function to get the offical, public or private items. The code here seems like just repeating the logic except some value differences.

Reply:
It is the jsx code of react code, works as view in MVC model. This part of code controls the UI rendering components in a hierarchical style. We could create function to work as rendering function, but it is prefered to put the rendering logic directly in the component for easy to read.

Although the code part is long but for me it is clear because of the HTML-like hierarchy, and maybe wrapping in a function could  not reduce the coding length.

Reply:
no problem. You can keep the current code. :smile: